pkg/daemon/ceph/client/upgrade_test.go

```diff
@@ -69,12 +82,90 @@ func TestEnableNautilusOSD(t *testing.T) {
 	executor.MockExecuteCommandWithOutput = func(debug bool, name string, command string, args ...string) (string, error) {
 		assert.Equal(t, "osd", args[0])
 		assert.Equal(t, "require-osd-release", args[1])
-		assert.Equal(t, "nautilus", args[2])
-		assert.Equal(t, 3, len(args))
 		return "", nil
 	}
 	context := &clusterd.Context{Executor: executor}
 
 	err := EnableNautilusOSD(context)
 	assert.Nil(t, err)
 }
+
+func TestOkToStopDaemon(t *testing.T) {
+	executor := &exectest.MockExecutor{}
+	executor.MockExecuteCommandWithOutput = func(debug bool, name string, command string, args ...string) (string, error) {
+		assert.Equal(t, "mon", args[0])
+		assert.Equal(t, "ok-to-stop", args[1])
+		assert.Equal(t, "a", args[2])
+		return "", nil
+	}
+	context := &clusterd.Context{Executor: executor}
+
+	deployment := "rook-ceph-mon-a"
+	err := okToStopDaemon(context, deployment, "rook-ceph")
+	assert.Nil(t, err)
+}
+
+func TestFindDaemonName(t *testing.T) {
+	n, err := findDaemonName("rook-ceph-mon-a")
+	assert.Nil(t, err)
+	assert.Equal(t, "mon", n)
+	n, err = findDaemonName("rook-ceph-osd-0")
+	assert.Nil(t, err)
+	assert.Equal(t, "osd", n)
+	n, err = findDaemonName("rook-ceph-rgw-my-store-a")
+	assert.Nil(t, err)
+	assert.Equal(t, "rgw", n)
+	n, err = findDaemonName("rook-ceph-mds-myfs-a")
+	assert.Nil(t, err)
+	assert.Equal(t, "mds", n)
+	n, err = findDaemonName("rook-ceph-mgr-a")
+	assert.Nil(t, err)
+	assert.Equal(t, "mgr", n)
+	n, err = findDaemonName("rook-ceph-rbd-mirror-a")
+	assert.Nil(t, err)
+	assert.Equal(t, "rbd-mirror", n)
+	_, err = findDaemonName("rook-ceph-unknown-a")
+	assert.NotNil(t, err)
+}
+
+func TestFindDaemonID(t *testing.T) {
+	id := findDaemonID("rook-ceph-mon-a")
+	assert.Equal(t, "a", id)
+	id = findDaemonID("rook-ceph-osd-0")
+	assert.Equal(t, "0", id)
+	id = findDaemonID("rook-ceph-rgw-my-super-store-a")
+	assert.Equal(t, "a", id)
+	id = findDaemonID("rook-ceph-mds-my-wonderful-fs-a")
+	assert.Equal(t, "a", id)
+	id = findDaemonID("rook-ceph-mgr-a")
+	assert.Equal(t, "a", id)
+	id = findDaemonID("rook.ceph.mgr.a")
+	assert.NotEqual(t, "a", id)
+}
+
+func TestOkToContinue(t *testing.T) {
```

Comment:
Add negative/error condition test.

Reply:
There is no real negative condition, if it's negative then it's skipped.

Reply:
There is no test to make sure this function returns error when it is supposed to.

Reply:
The only that can happen is the under-lying function that fails, I'm not sure how to mock that.